lib/taskana-core/src/test/java/acceptance/task/CancelTaskAccTest.java

```diff
@@ -0,0 +1,128 @@
+package acceptance.task;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import acceptance.AbstractAccTest;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import pro.taskana.common.api.exceptions.NotAuthorizedException;
+import pro.taskana.security.JaasExtension;
+import pro.taskana.security.WithAccessId;
+import pro.taskana.task.api.TaskService;
+import pro.taskana.task.api.TaskState;
+import pro.taskana.task.api.exceptions.InvalidStateException;
+import pro.taskana.task.api.exceptions.TaskNotFoundException;
+import pro.taskana.task.api.models.Task;
+import pro.taskana.task.api.models.TaskSummary;
+
+
+/** Acceptance tests for all claim and complete scenarios. */
+@ExtendWith(JaasExtension.class)
+class CancelTaskAccTest extends AbstractAccTest {
+  private TaskService taskService;
+
+  CancelTaskAccTest() {
+    super();
+    taskService = taskanaEngine.getTaskService();
+  }
+
+  @BeforeEach
+  public static void setupTest() throws Exception {
+    resetDb(false);
+  }
+
+  @WithAccessId(
+      userName = "user_1_1",
+      groupNames = {"group_1"})
+  @Test
+  void testQeryCancelledTasks() {
+    List<TaskSummary> taskSummaries =
+        taskService.createTaskQuery().stateIn(TaskState.CANCELLED).list();
+    assertThat(taskSummaries.size()).isEqualTo(5);
+  }
+
+  @WithAccessId(
+      userName = "admin",
+      groupNames = {"group_1"})
+  @Test
+  void testCancelReadyTask()
+      throws NotAuthorizedException, TaskNotFoundException, InvalidStateException {
+    List<TaskSummary> taskSummaries = taskService.createTaskQuery().stateIn(TaskState.READY).list();
+    assertThat(taskSummaries.size()).isEqualTo(47);
+    Task task = taskService.getTask(taskSummaries.get(0).getId());
+    taskService.cancelTask(taskSummaries.get(0).getId());
+    long numTasks = taskService.createTaskQuery().stateIn(TaskState.READY).count();
+    assertThat(numTasks).isEqualTo(46);
+    numTasks = taskService.createTaskQuery().stateIn(TaskState.CANCELLED).count();
+    assertThat(numTasks).isEqualTo(6);
+  }
+
+  @WithAccessId(
+      userName = "admin",
```

Comment:
Did we decide that everybody is allowed to cancel a claimed task? I wonder if it would be better to allow only the owner to cancel claimed tasks.

Reply:
The ticket says : " terminateTask() and cancelTask bay be called by every user that is allowed to see the task " 
I agree, that it is strange that everybody is allowed to cancel / terminate a task. On the other hand, if you think about the adapter scenario: if a task has been cancelled / terminated in camunda, the adapter should be able to cancel / terminate it in taskana. And this without being the owner. And probably also without having admin authority. So before I change anything with respect to this, we should carefully think about what is the correct solution ....